### PR TITLE
Fix to make browser back button navigate pagination properly when faceted search is enabled

### PR DESCRIPTION
**Pull request by bc-williamkwon**
#### What?
Tested on store: https://williamkwon1576004500-testly-the-third.my-integration.zone/
Previously, the browser back button would not navigate pagination properly when faceted search was enabled but now it does.

#### Tickets / Documentation

Add links to any relevant tickets and documentation.

- [Link 1](https://jira.bigcommerce.com/browse/STRF-7232)

**Screenshots (if appropriate)**
https://imgur.com/a/MtmIeIR

### CODE REVIEW

**Comment by mattolson** [comment]
Please provide proof that the change works as expected.

**Comment by bc-williamkwon** [comment]
Added a screenshot and linked to store with theme that has changes

**Comment by bookernath** [comment]
💚

**Comment by bookernath** [comment]
I see the issue is still happening with a filter applied, e.g.:

https://williamkwon1575504304-testsworthy.my-integration.zone/shop-all/?_bc_fsnf=1&brand=36&limit=2

From here, you can navigate to page 2 and then click the back button - you'll notice the page number in the content footer does not change, and the products do not update.

**Comment by bc-williamkwon** [comment]
Made changes to fix issues.

**Comment by bookernath** [comment]
💚 all cases seem fixed now. Nice work!

**Comment by bookernath** [comment]
The issue with the page not reloading seems fixed now.

However, it appears that we've introduced a new bug with this change:

Now if I'm on page 1 and I directly click on page 3, and then click the back button, I'll be taken to page 2, even though I never visited that page so it shouldn't be in my history.

**Comment by bc-williamkwon** [comment]
Fixed said bug described in last comment

File: assets/js/theme/common/faceted-search.js [hunk 1]
```diff
@@ -408,6 +410,18 @@ class FacetedSearch {
...
         }
     }
+
+    onPopState() {
+        const currentUrl = window.location.href;
+        const searchParams = new URLSearchParams(currentUrl);
+        // If searchParams does not contain a page value then modify url query string to have page=1
+        if (!searchParams.has('page')) {
+            const linkUrl = $('.pagination-link').attr('href');
+            const url = `${linkUrl.slice(0, linkUrl.indexOf('page'))}page=1`;
```
Comment (mattolson) [inline assets/js/theme/common/faceted-search.js]: Why does `linkUrl.indexOf('page')` return something if `searchParams.has('page')` is false?

Comment (bc-williamkwon) [reply assets/js/theme/common/faceted-search.js]: linkUrl is taken from the dom searchParams is taken from the url.  Each page has a corresponding href on the dom.

Comment (mattolson) [reply assets/js/theme/common/faceted-search.js]: I think we need to use something more robust that updates just the page parameter. `linkUrl` could contain other search parameters when facets are selected, and this code assumes that `page` is the last parameter, which may not always be the case.

**Comment by bc-williamkwon** [comment]
Made change to only update page parameter

**Comment by bc-williamkwon** [comment]
Updated solution to use regex instead of split.

**Comment by bookernath** [comment]
I've tested all the cases I tested before, and this looks good now 💚

**Pull request merged**